mlir/include/mlir/TableGen/Pattern.h

```diff
@@ -279,6 +299,12 @@ class SymbolInfoMap {
     //   the DAG of the operation, `operandIndexOrNumValues` specifies the
     //   operand index, and `variadicSubIndex` must be set to `std::nullopt`.
     //
+    // * Properties not associated with an operation (ex. as arguments to
+    //   native code) have their corresponding PropConstraint stored in the
+    //   `dag` field,
+    //  and set `operandIndexOrNumValues` to -1 to indicate they aren't part of
```

Comment:
Does the user of this interface need to know about the sentinel or is this internal only?

Reply:
Moved the sentinel to a parenthetical